packages/front-end/components/Settings/WebhooksModal.tsx

```diff
@@ -61,13 +64,29 @@ const WebhooksModal: FC<{
     }
   };
 
+  const isValidHttp = (urlString: string) => {
+    let url;
+    try {
+      url = new URL(urlString);
+    } catch (e) {
+      return false;
+    }
+    return /https?/.test(url.protocol);
+  };
+
   const onSubmit = form.handleSubmit(async (value) => {
+    setDisplayHttpValidationError(false);
     if (value.endpoint.match(/localhost/g)) {
       throw new Error("Invalid endpoint");
     }
+    if (!isValidHttp(value.endpoint)) {
+      setDisplayHttpValidationError(true);
+      return;
+    }
     await handleApiCall(value);
     track(current.id ? "Edit Webhook" : "Create Webhook");
     onSave();
+    close();
```

Comment:
If you make the above change, you can get rid of this and the `autoCloseOnSubmit` change